server/modules/analysis/service/_rChain/rChain.js

```diff
@@ -89,6 +91,8 @@ class RChain {
 
   async _initSurveyAndChain() {
     this._survey = await SurveyManager.fetchSurveyAndNodeDefsBySurveyId(this.surveyId, this.cycle)
+    const categories = await CategoryManager.fetchCategoriesAndLevelsBySurveyId(this.surveyId)
+    this._survey = Survey.assocCategories(categories)(this.survey)
```

Comment:
why do you fetch all categories? isn't it enough to fetch only the category that is needed in rFileReadData?

Reply:
because both rFileReadData and rFilePersistResults need to fetch the category... is it easier to fetch them at the beginning?

Reply:
ok!